components/query/query-wms.service.ts

```diff
@@ -1,72 +1,103 @@
+import {HttpClient} from '@angular/common/http';
 import {Image as ImageLayer, Tile} from 'ol/layer';
 import {ImageWMS} from 'ol/source';
+import {Injectable} from '@angular/core';
 import {TileWMS} from 'ol/source';
 
-/**
- * @param $rootScope
- * @param $http
- * @param HsQueryBaseService
- * @param HsMapService
- * @param HsUtilsService
- * @param HsLanguageService
- * @param HsLayerUtilsService
- */
-export default function (
-  $rootScope,
-  $http,
-  HsQueryBaseService,
-  HsMapService,
-  HsUtilsService,
-  HsLanguageService,
-  HsLayerUtilsService
-) {
-  'ngInject';
-  const me = this;
-
-  me.infoCounter = 0;
-
-  this.request = function (url, infoFormat, coordinate, layer) {
-    const req_url = HsUtilsService.proxify(url, true);
-    const reqHash = HsQueryBaseService.currentQuery;
-    $http({url: req_url})
-      .then((response) => {
-        if (reqHash != HsQueryBaseService.currentQuery) {
+@Injectable({
+  providedIn: 'root',
+})
+export class HsQueryWmsService {
+  infoCounter = 0;
+  constructor(
+    private HsQueryBaseService: HsQueryBaseService,
+    private HsMapService: HsMapService,
+    private HsConfig: HsConfig,
+    private HsLayerUtilsService: HsLayerUtilsService,
+    private HsLanguageService: HsLanguageService,
+    private HsUtilsService: HsUtilsService,
+    private HsEventBusService: HsEventBusService,
+    private HttpClient: HttpClient
+  ) {
+    this.HsQueryBaseService.getFeatureInfoStarted.subscribe((evt) => {
+      this.infoCounter = 0;
+      this.HsMapService.map.getLayers().forEach((layer) => {
+        if (layer.get('base') == true || layer.get('queriable') == false) {
           return;
         }
-        me.featureInfoReceived(
-          response.data,
-          infoFormat,
-          url,
-          coordinate,
-          layer
-        );
-      })
-      .catch((err) => {
-        if (reqHash != HsQueryBaseService.currentQuery) {
-          return;
+        if (layer.get('queryFilter') != undefined) {
+          const filter = layer.get('queryFilter');
+          if (filter(HsMapService.map, layer, evt.pixel)) {
+            this.queryWmsLayer(layer, evt.coordinate);
+          }
+        } else {
+          this.queryWmsLayer(layer, evt.coordinate);
         }
-        me.featureInfoError(coordinate);
       });
-  };
+    });
+  }
+
+  /**
+   * @param updated
+   * @param customInfoTemplate
+   * @param Base
+   * @param group
+   */
+  updateFeatureList(updated, customInfoTemplate, Base, group) {
+    if (updated) {
+      if (customInfoTemplate) {
+        Base.setData(group, 'customFeatures');
+        Base.dataCleared = false;
+      } else {
+        Base.setData(group, 'features');
+      }
+    }
+  }
+
+  async request(url, infoFormat, coordinate, layer) {
+    const req_url = this.HsUtilsService.proxify(url, true);
+    const reqHash = this.HsQueryBaseService.currentQuery;
+    try {
+      const response = await this.HttpClient.get(req_url).toPromise();
```

Comment:
Tried getFeatureInfo request with this WMS: http://geoportal.gov.cz/arcgis/services/CENIA/cenia_corine_2012/MapServer/WmsServer?.
Request was made but the response was not processed as there was no `featureInfoReceived()` call. 

Only managed to get proper response that appears in info panel by replacing toPromise with subscribe. Also XML headers were necessary then
```
    try {
      const headers = new Headers({'Content-Type': 'text/xml'});
      headers.set('Accept', 'text/xml');
      headers.set('Content-Type', 'text/xml');
      await this.HttpClient.get(req_url, {
        headers: new HttpHeaders().set('Content-Type', 'text/xml'),
        responseType: 'text',
      }).subscribe((response) => {
        if (reqHash != this.HsQueryBaseService.currentQuery) {
          return;
        }
        this.featureInfoReceived(response, infoFormat, url, coordinate, layer);
      });
    } 
```

Reply:
Thanks @FilipLeitner for giving me some pointers. I changed it to {'Content-Type': 'text'} to support wider range of info_formats. Also toPromise() works for me now.